File: backend/src/admin/objects/PlatformMetaResponse.scala
package delivery.admin.objects

final case class PlatformMetaResponse(
    campaigns: List[PromotionCampaign],
    complaintTickets: List[ComplaintTicket],
    merchantApplications: List[MerchantApplication],
    serviceAgents: List[CustomerServiceAgent],
    operationsManagers: List[OperationsManager]
)

File: frontend/src/admin/objects/PlatformMetaResponse.ts
import type { ComplaintTicket } from './ComplaintTicket'
import type { CustomerServiceAgent } from './CustomerServiceAgent'
import type { MerchantApplication } from './MerchantApplication'
import type { OperationsManager } from './OperationsManager'
import type { PromotionCampaign } from './PromotionCampaign'

export interface PlatformMetaResponse {
  campaigns: PromotionCampaign[]
  complaintTickets: ComplaintTicket[]
  merchantApplications: MerchantApplication[]
  serviceAgents: CustomerServiceAgent[]
  operationsManagers: OperationsManager[]
}
